src/datasets.js

```diff
@@ -75,37 +76,60 @@ import * as Utils from './utils'; // eslint-disable-line import/no-duplicates
  * @namespace datasets
  */
 
-export const BASE_PATH = '/v2/datasets';
 export const SIGNED_URL_UPLOAD_MIN_BYTESIZE = 1024 * 256;
 
-export default {
+export default class Datasets {
+    constructor(httpClient) {
+        this.basePath = '/v2/datasets';
+        this.client = httpClient;
+    }
+
+    _call(userOptions, endpointOptions) {
```

Comment:
This method is here 4 times. You could make a default abstract class and inherit from it